api/_dexes/cross-swap.ts

```diff
@@ -242,23 +241,39 @@ export async function buildCrossSwapTx(
     toAddress = spokePool.address;
   }
 
-  const [gas, gasPrice] = await Promise.all([
-    spokePool.provider.estimateGas({
-      from: crossSwapQuotes.crossSwap.depositor,
-      ...tx,
-    }),
-    latestGasPriceCache(originChainId).get(),
-  ]);
-
   return {
     from: crossSwapQuotes.crossSwap.depositor,
     to: toAddress,
     data: integratorId ? tagIntegratorId(integratorId, tx.data!) : tx.data,
-    gas,
-    gasPrice,
     value: tx.value,
   };
 }
+async function extractDepositDataStruct(crossSwapQuotes: CrossSwapQuotes) {
+  const originChainId = crossSwapQuotes.crossSwap.inputToken.chainId;
+  const destinationChainId = crossSwapQuotes.crossSwap.outputToken.chainId;
+  const spokePool = getSpokePool(originChainId);
+  const message = crossSwapQuotes.bridgeQuote.message || "0x";
+  const deposit = {
+    depositor: crossSwapQuotes.crossSwap.depositor,
+    recipient:
+      message && message !== "0x"
+        ? getMultiCallHandlerAddress(destinationChainId)
+        : crossSwapQuotes.crossSwap.recipient,
+    inputToken: crossSwapQuotes.bridgeQuote.inputToken.address,
+    outputToken: crossSwapQuotes.bridgeQuote.outputToken.address,
+    inputAmount: crossSwapQuotes.bridgeQuote.inputAmount,
+    outputAmount: crossSwapQuotes.bridgeQuote.outputAmount,
+    destinationChainid: crossSwapQuotes.bridgeQuote.outputToken.chainId,
```

Comment:
```suggestion
    destinationChainid: crossSwapQuotes.bridgeQuote.outputToken.chainId,
```

We also have `destinationChainId = crossSwapQuotes.crossSwap.outputToken.chainId`. Is it worth asserting that these are equal? Then we could just use `destinationChainId` here.